.github/workflows/ci.yml

```diff
@@ -0,0 +1,73 @@
+name: CI
+
+on: [pull_request, push]
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-latest
+            artifact-name: Linux
+            architecture: x64
+            build-options: ""
+          - os: ubuntu-latest
+            artifact-name: Linux-cuda
+            architecture: x64
+            build-options: "-Pcuda -x test"
+          - os: windows-latest
+            artifact-name: Win64
+            architecture: x64
+            build-options: "-PskipUITests"
+          - os: windows-latest
+            artifact-name: Win64-cuda
+            architecture: x64
+            build-options: "-PskipUITests -Pcuda -x test"
+          - os: windows-latest
+            artifact-name: Win32
+            architecture: x86
+            build-options: "-PskipUITests"
+          - os: macos-latest
+            artifact-name: macOS
+            architecture: x64
+            build-options: ""
+    name: "Build - ${{ matrix.artifact-name }}"
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: joschi/setup-jdk@v2
+        id: setup-java-14
+        with:
+          java-version: 14
+          architecture: ${{ matrix.architecture }}
+      - uses: joschi/setup-jdk@v2
```

Comment:
Why not use the official one? Why use this one instead?
https://github.com/actions/setup-java

Reply:
This way it matches exactly what Azure had. The official one uses Azul JDK